course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineViewModel.kt

```diff
@@ -121,6 +133,24 @@ class CourseOutlineViewModel(
         getCourseDataInternal()
     }
 
+    fun switchCourseSections(blockId: String): Boolean {
+        courseSectionsState[blockId] = !(courseSectionsState[blockId] ?: false)
+        if (_uiState.value is CourseOutlineUIState.CourseData) {
+            val state = _uiState.value as CourseOutlineUIState.CourseData
+            _uiState.value = null
+            _uiState.value = CourseOutlineUIState.CourseData(
+                courseStructure = state.courseStructure,
+                downloadedState = state.downloadedState,
+                resumeComponent = state.resumeComponent,
+                courseSections = courseSections,
+                courseSectionsState = courseSectionsState,
+                downloadsCount = downloadsCount
+            )
```

Comment:
What is the purpose of this workaround?
Can we remove `_uiState.value = null`?

Reply:
Hi! I changed how states for expandable items work and removed this line. Thank you!

Reply:
Thank you!